src/App.tsx

```diff
@@ -1,34 +1,61 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { todosActions } from './features/todos';
 
 export const App: React.FC = () => {
+  const [loading, setLoading] = useState(true);
+  const [hasError, setHasError] = useState(false);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const todos = await getTodos();
+
+        dispatch(todosActions.loadTodos(todos));
+      } catch {
+        setHasError(true);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetchData();
+  }, []);
+
   return (
     <>
       <div className="section">
         <div className="container">
           <div className="box">
             <h1 className="title">Todos:</h1>
 
+            {loading && <Loader />}
+
             <div className="block">
               <TodoFilter />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {hasError
+                ? <span> No todos from server</span>
+                : <TodoList />}
```

Comment:
This way your `<TodoList />` will be shown even if `loading` is true. I don't think it's a correct behaviour.